pupil-auth/app.js

```diff
@@ -0,0 +1,169 @@
+'use strict'
+
+require('dotenv').config()
+
+const express = require('express')
+const piping = require('piping')
+const path = require('path')
+const morgan = require('morgan')
+const uuidV4 = require('uuid/v4')
+const cors = require('cors')
+const helmet = require('helmet')
+const config = require('./config')
+const devWhitelist = require('./whitelist-dev')
+const azure = require('./azure')
+const winston = require('winston')
+const bodyParser = require('body-parser')
+
+if (process.env.NODE_ENV !== 'production') {
+  winston.level = 'debug'
+}
+
+const unsetVars = []
+Object.keys(config).map((key) => {
+  if (config[key] === undefined && !devWhitelist.includes(key)) {
+    unsetVars.push(`${key}`)
+  }
+})
+
+if (unsetVars.length > 0) {
+  const error = `The following environment variables need to be defined:\n${unsetVars.join('\n')}`
+  process.exitCode = 1
+  throw new Error(error)
+}
+
+const index = require('./routes/index')
+
+if (process.env.NODE_ENV === 'development') piping({ignore: [/test/, '/coverage/']})
+const app = express()
+
+if (config.Logging.Express.UseWinston === 'true') {
+  /**
+   * Express logging to winston
+   */
+  const expressWinston = require('express-winston')
+  app.use(expressWinston.logger({
+    transports: [
+      new winston.transports.Console({
+        json: true,
+        colorize: true
+      })
+    ],
+    meta: true, // optional: control whether you want to log the meta data about the request (default to true)
+    // msg: "HTTP {{req.method}} {{req.url}}", // optional: customize the default logging message. E.g. "{{res.statusCode}} {{req.method}} {{res.responseTime}}ms {{req.url}}"
+    expressFormat: true, // Use the default Express/morgan request formatting. Enabling this will override any msg if true. Will only output colors with colorize set to true
+    colorize: false, // Color the text and status code, using the Express/morgan color palette (text: gray, status: default green, 3XX cyan, 4XX yellow, 5XX red).
+    ignoreRoute: function (req, res) {
+      return false
+    } // optional: allows to skip some log messages based on request and/or response
+  }))
+} else {
+  app.use(morgan('dev'))
+}
+
+/* Security Directives */
+
+app.use(cors())
+app.use(helmet())
+const scriptSources = ["'self'", "'unsafe-inline'"]
+const styleSources = ["'self'", "'unsafe-inline'"]
+const imgSources = ["'self'", 'data:']
+const objectSources = ["'self'"]
+
+if (config.AssetPath !== '/') {
+  // add CSP policy for assets domain
+  scriptSources.push(config.AssetPath)
+  styleSources.push(config.AssetPath)
+  imgSources.push(config.AssetPath)
+  objectSources.push(config.AssetPath)
+}
+app.use(helmet.contentSecurityPolicy({
```

Comment:
I suspect that this security policy can be reset to nothing.   It's just an API end-point isn't it?